core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -447,6 +449,11 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     _conf.set("spark.externalBlockStore.folderName", externalBlockStoreFolderName)
 
     if (master == "yarn-client") System.setProperty("SPARK_YARN_MODE", "true")
+    if (CryptoConf.isShuffleEncrypted(_conf)) {
+      if (!SparkHadoopUtil.get.isYarnMode()) {
+        logError("Shuffle encryption is only supported in Yarn mode!")
```

Comment:
So, you probably want something stronger than just a log. Either:

- Fix the configuration so that encryption is disabled (and say that in the log message).
- Throw an exception.